Lib/test/test_importlib/import_/test_path.py

```diff
@@ -153,6 +154,25 @@ def test_deleted_cwd(self):
             # Do not want FileNotFoundError raised.
             self.assertIsNone(self.machinery.PathFinder.find_spec('whatever'))
 
+    @os_helper.skip_unless_working_chmod
+    def test_permission_error_cwd(self):
+        # gh-115911
+        with (
+            os_helper.temp_dir() as new_dir,
+            os_helper.save_mode(new_dir),
+            os_helper.save_cwd(),
+            util.import_state(path=['']),
+        ):
+            os.chdir(new_dir)
```

Comment:
Why not re-use `change_cwd`?

```suggestion
        with (
            os_helper.temp_dir() as new_dir,
            os_helper.save_mode(new_dir),
            os_helper.change_cwd(new_dir),
            util.import_state(path=['']),
        ):
```

Consider also `save_mode` could be `change_mode` (except it's maybe not possible to implement the skip behavior).